ssh/proxy.go

```diff
@@ -55,7 +55,7 @@ func (p *ProxyConn) handleAuthMsg(msg *userAuthRequestMsg, proxyConf *ProxyConfi
 			proxyConf.FetchAuthorizedKeysHook = fetchAuthorizedKeysFromHomeDir
 		}
 
-		downStreamPublicKey, isQuery, sig, err := parsePublicKeyMsg(msg)
+		downStreamPublicKey, isQuery, sig, algo, err := parsePublicKeyMsg(msg)
```

Comment:
rsa public key type is not same with algo no more, so signature verify must use algo string.

ex)

```
algo: rsa-sha2-512, pubkey type: ssh-rsa
```

ssh-ed25519 type is same, but it's not a problem for verifying 😓 

```
algo: ssh-ed25519, pubkey type: ssh-ed25519
```